src/plugins/credential_management/common/types.ts

```diff
@@ -9,23 +9,29 @@
  * GitHub history for details.
  */
 
-// TODO: refactor the credential in heritance:
-export interface ICredential {
-  readonly credential_name: string;
-  readonly credential_type: CredentialType;
-  readonly credential_material: IUserNamePasswordCredentialMaterial | IAWSIAMCredentialMaterial;
+import { SavedObjectAttributes } from '../../../../src/core/types';
+
+export type UserNamePasswordType = 'username_password_credential';
+export type AWSIAMType = 'aws_iam_credential';
+
+export interface CredentialSavedObjectAttributes extends SavedObjectAttributes {
+  title: string;
+  credentialType: string;
```

Comment:
let's define strong type for credentials type, arbitrary string is hard to verify

Reply:
The credential type in this case differs from the credential materials type. As part of the product discussion, I would begin with the "Shared" type and then add more later.